package chapters.chapter_12.Checkpoints12;

public class Checkpoint12_24 {
   /*
    public static void main(String[] args) {
        try{
            method1();
        }
        catch (Exception ex){
            ex.printStackTrace();
        }
    }

    public static void method1() throws Exception{
        try{
            method2();
        }
        catch (Exception ex){
            throw new Exception("New info from method1", ex);
        }
    }
    public static void method2() throws Exception{
        throw new Exception("New info from method2");
}
Output:
java.lang.Exception: New info from method1
	at chapters.chapter_12.Listings12.Listing12_09_ChainedExceptionDemo.method1(Listing12_09_ChainedExceptionDemo.java:18)
	at chapters.chapter_12.Listings12.Listing12_09_ChainedExceptionDemo.main(Listing12_09_ChainedExceptionDemo.java:6)

*/
}